addons/account_audit_trail/models/mail_message.py

```diff
@@ -43,32 +69,116 @@ def _compute_account_audit_log_preview(self):
+            ('check_account_audit_trail', '=', True),
+        ])
+
+    def _search_account_audit_log_company_id(self, operator, value):
+        return self._search_audit_log_related_record_id('res.company', operator, value)
+
+    def _compute_account_audit_log_partner_id(self):
+        self._compute_audit_log_related_record_id('res.partner', 'account_audit_log_partner_id', [
+            '|', ('company_id', '=', False), ('company_id.check_account_audit_trail', '=', True),
+            '|', ('customer_rank', '>', 0), ('supplier_rank', '>', 0),
+        ])
+
+    def _search_account_audit_log_partner_id(self, operator, value):
+        return self._search_audit_log_related_record_id('res.partner', operator, value)
+
+    def _compute_account_audit_log_display_name(self):
+        for message in self:
+            message.account_audit_log_display_name = (
+                message.account_audit_log_move_id
+                or message.account_audit_log_account_id
+                or message.account_audit_log_tax_id
+                or message.account_audit_log_partner_id
+                or message.account_audit_log_company_id
+            ).display_name
+
+    def _compute_show_audit_log(self):
+        for message in self:
+            message.show_audit_log = message.message_type == 'notification' and (
+                message.account_audit_log_move_id
+                or message.account_audit_log_account_id
+                or message.account_audit_log_tax_id
+                or message.account_audit_log_partner_id
+                or message.account_audit_log_company_id
+            )
+
+    def _search_show_audit_log(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise UserError(_('Operation not supported'))
+        return [('message_type', '=', 'notification')] + OR([
+            [('model', '=', 'account.move'), ('res_id', 'in', self.env['account.move']._search([
+                ('company_id.check_account_audit_trail', operator, value),
+            ]))],
+            [('model', '=', 'account.account'), ('res_id', 'in', self.env['account.account']._search([
+                ('company_id.check_account_audit_trail', operator, value),
+            ]))],
+            [('model', '=', 'account.tax'), ('res_id', 'in', self.env['account.tax']._search([
+                ('company_id.check_account_audit_trail', operator, value),
+            ]))],
+            [('model', '=', 'res.partner'), ('res_id', 'in', self.env['res.partner']._search([
+                '|', ('company_id', '=', False), ('company_id.check_account_audit_trail', operator, value),
+                '|', ('customer_rank', '>', 0), ('supplier_rank', '>', 0),
+            ]))],
+            [('model', '=', 'res.company'), ('res_id', 'in', self.env['res.company']._search([
+                ('check_account_audit_trail', operator, value),
+            ]))],
+        ])
```

Comment:
Purely personal opinion on indentation. I feel like this suggestion is cleaner and more readable:
```suggestion
        return [('message_type', '=', 'notification')] + OR([
            [
                ('model', '=', 'account.move'),
                ('res_id', 'in', self.env['account.move']._search([
                    ('company_id.check_account_audit_trail', operator, value),
                ])),
            ],
            [
                ('model', '=', 'account.account'),
                ('res_id', 'in', self.env['account.account']._search([
                    ('company_id.check_account_audit_trail', operator, value),
                ])),
            ],
            [
                ('model', '=', 'account.tax'),
                ('res_id', 'in', self.env['account.tax']._search([
                    ('company_id.check_account_audit_trail', operator, value),
                ])),
            ],
            [
                ('model', '=', 'res.partner'),
                ('res_id', 'in', self.env['res.partner']._search([
                    '|', ('company_id', '=', False), ('company_id.check_account_audit_trail', operator, value),
                    '|', ('customer_rank', '>', 0), ('supplier_rank', '>', 0),
                ])),
            ],
            [
                ('model', '=', 'res.company'),
                ('res_id', 'in', self.env['res.company']._search([
                    ('check_account_audit_trail', operator, value),
                ])),
            ],
        ])
```

Do what you want with it :nerd_face:

Reply:
I would agree if these domains were likely to change, but it is not the case